Update to v1.3.4 in Ubuntu

lucaskanashiro opened the pull request:
This PR contains the following:

* Import version 1.3.4
* Update Vcs urls to point to this git repository
* Remove a patch applied by upstream in version 1.3.4
* Add a patch to fix a FTBFS against Go 1.14 (default in Groovy)

The proposed packages is available in this PPA:

https://launchpad.net/~lucaskanashiro/+archive/ubuntu/groovy-containerd-merge/+packages

The DEP-8 test is passing as expected:

```
autopkgtest [17:03:57]: test basic-smoke: [-----------------------
+ ctr images pull docker.io/library/busybox:latest
+ containerd
time="2020-05-14T17:03:58.328978663-03:00" level=info msg="starting containerd" revision= version=1.3.4-0ubuntu1
time="2020-05-14T17:03:58.352701768-03:00" level=info msg="loading plugin \"io.containerd.content.v1.content\"..." type=io.containerd.content.v1
time="2020-05-14T17:03:58.352788397-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.353052590-03:00" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." error="path /var/lib/containerd/io.containerd.snapshotter.v1.btrfs (1
time="2020-05-14T17:03:58.353068675-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.devmapper\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.353083016-03:00" level=warning msg="failed to load plugin io.containerd.snapshotter.v1.devmapper" error="devmapper not configured"
time="2020-05-14T17:03:58.353091528-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.aufs\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.355907183-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.native\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.355937511-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.overlayfs\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.355996600-03:00" level=info msg="loading plugin \"io.containerd.snapshotter.v1.zfs\"..." type=io.containerd.snapshotter.v1
time="2020-05-14T17:03:58.356149682-03:00" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.zfs\"..." error="path /var/lib/containerd/io.containerd.snapshotter.v1.zfs must 1
time="2020-05-14T17:03:58.356162320-03:00" level=info msg="loading plugin \"io.containerd.metadata.v1.bolt\"..." type=io.containerd.metadata.v1
time="2020-05-14T17:03:58.356176045-03:00" level=warning msg="could not use snapshotter devmapper in metadata plugin" error="devmapper not configured"
time="2020-05-14T17:03:58.356184029-03:00" level=info msg="metadata content store policy set" policy=shared
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.1 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.2 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.3 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.4 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.5 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.6 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.7 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.8 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 0.9 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.0 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.1 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.2 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.3 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.4 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.5 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.6 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.7 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.8 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest: resolving      |--------------------------------------| 
elapsed: 1.9 s                    total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest:                                              resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c: downloading    |--------------------------------------|    0.0 B/2.0 KiB 
elapsed: 2.0 s                                                                 total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest:                                              resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c: downloading    |--------------------------------------|    0.0 B/2.0 KiB 
elapsed: 2.1 s                                                                 total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest:                                              resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c: downloading    |--------------------------------------|    0.0 B/2.0 KiB 
elapsed: 2.2 s                                                                 total:   0.0 B (0.0 B/s)                                         
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: downloading    |--------------------------------------|    0.0 B/527.0 B 
elapsed: 2.3 s                                                                    total:  2.0 Ki (903.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: downloading    |--------------------------------------|    0.0 B/527.0 B 
elapsed: 2.4 s                                                                    total:  2.0 Ki (865.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: downloading    |--------------------------------------|    0.0 B/527.0 B 
elapsed: 2.5 s                                                                    total:  2.0 Ki (831.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 2.6 s                                                                    total:  2.5 Ki (1002.0 B/s)                                      
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 2.7 s                                                                    total:  2.5 Ki (964.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 2.8 s                                                                    total:  2.5 Ki (929.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 2.9 s                                                                    total:  2.5 Ki (897.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 3.0 s                                                                    total:  2.5 Ki (867.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |--------------------------------------|    0.0 B/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   downloading    |--------------------------------------|    0.0 B/1.5 KiB   
elapsed: 3.1 s                                                                    total:  2.5 Ki (839.0 B/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |+-------------------------------------| 23.1 KiB/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.2 s                                                                    total:  27.1 K (8.5 KiB/s)                                       
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |++++++++------------------------------| 163.2 Ki/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.3 s                                                                    total:  167.2  (50.7 KiB/s)                                      
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |++++++++++++--------------------------| 239.1 Ki/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.4 s                                                                    total:  243.1  (71.4 KiB/s)                                      
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |++++++++++++++------------------------| 287.1 Ki/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.5 s                                                                    total:  291.1  (83.1 KiB/s)                                      
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |++++++++++++++++++++++++++------------| 511.1 Ki/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.6 s                                                                    total:  515.1  (143.0 KiB/s)                                     
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    downloading    |+++++++++++++++++++++++++++++++-------| 607.1 Ki/743.0 KiB 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.7 s                                                                    total:  611.1  (165.0 KiB/s)                                     
docker.io/library/busybox:latest:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:a7766145a775d39e53a713c75b6fd6d318740e70327aaa3ed5d09e0ef33fc3df: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d9cbbca60e5f0fc028b13c01e70e96cc2eefe7858ccc5a33c9a8faa8b03286aa:    done           |++++++++++++++++++++++++++++++++++++++| 
config-sha256:78096d0a54788961ca68393e5f8038704b97d8af374249dc5c8faec1b8045e42:   done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 3.8 s                                                                    total:  611.1  (160.8 KiB/s)                                     
unpacking linux/amd64 sha256:836945da1f3afe2cfff376d379852bbb82e0237cb2925d53a13f53d6e8a8c48c...
++ ctr run --rm docker.io/library/busybox:latest test-1178-4344 echo success
done
+ output=success
+ '[' success = success ']'
autopkgtest [17:04:03]: test basic-smoke: -----------------------]
autopkgtest [17:04:04]: test basic-smoke:  - - - - - - - - - - results - - - - - - - - - -
basic-smoke          PASS
autopkgtest [17:04:04]: @@@@@@@@@@@@@@@@@@@@ summary
basic-smoke          PASS
```

lucaskanashiro committed:
Add patch to update a vendorized dep, fixes a FTBFS against Go 1.14

go.etcd.io/bbolt needs to be updated to version 1.3.4 in order to fix a
the following error against Go 1.14:

fatal error: checkptr: unsafe pointer conversion

This patch was backported from the upstream master branch, so it should
be removed in the next upstream release.

lucaskanashiro committed:
Update changelog

lucaskanashiro committed:
Update Vcs-{Git,Browser} to point to Github

lucaskanashiro committed:
Remove patch already applied by upstream

tianon commented on debian/changelog:
```diff
@@ -1,3 +1,14 @@
+containerd (1.3.4-0ubuntu1) groovy; urgency=medium
```
This should be `UNRELEASED` since this isn't uploaded yet:

```suggestion
containerd (1.3.4-0ubuntu1) UNRELEASED; urgency=medium
```

tianon commented on debian/control:
```diff
@@ -13,8 +13,8 @@ Build-Depends: debhelper (>= 9),
 Homepage: https://containerd.io
-Vcs-Git: https://anonscm.debian.org/git/pkg-go/packages/containerd.git
-Vcs-Browser: https://anonscm.debian.org/cgit/pkg-go/packages/containerd.git
+Vcs-Git: https://github.com/tianon/debian-containerd.git
+Vcs-Browser: https://github.com/tianon/debian-containerd
```
If we're updating these, we should probably include the branch, right?

```suggestion
Vcs-Git: https://github.com/tianon/debian-containerd.git -b ubuntu
Vcs-Browser: https://github.com/tianon/debian-containerd/tree/ubuntu
```

tianon commented on debian/patches/update_go.etcd.io_bbolt_to_v1.3.4.patch:
```diff
@@ -0,0 +1,884 @@
+From fb9e3d9f21074e60e9b74034dca5ef19f3456c4c Mon Sep 17 00:00:00 2001
+From: Sebastiaan van Stijn <github@gone.nl>
+Date: Mon, 30 Mar 2020 11:15:37 +0200
+Subject: [PATCH] vendor: update go.etcd.io/bbolt v1.3.4
```
I think we should probably include `Origin:` in here explicitly pointing to https://github.com/containerd/containerd/pull/4134, and I typically name upstream patches like this by their PR number so it's more obvious at a glance that they're direct-from-upstream, so I'd probably name this something like `4134-update-etcd-bbolt.patch`.

(Makes it easier to know which patches have a chance of being applied upstream on the next release and which might need refreshing instead.)

tianon reviewed (commented):
Generally looks good, thanks!  Just a few minor suggestions.

cc @mwhudson

lucaskanashiro commented:
@tianon I addressed all the points you raised.

lucaskanashiro committed:
Disable btrfs plugin on riscv64

As this plugin needs cgo and riscv64 doesn't support. These changes are
based on this Debian commit:

https://salsa.debian.org/go-team/packages/containerd/-/commit/e3c6a7bb

This is the upstream bug report:

https://github.com/golang/go/issues/36641

lucaskanashiro commented:
After checking these [Debian changes](https://salsa.debian.org/go-team/packages/containerd/-/commit/e3c6a7bb) and this [upstream bug report](https://github.com/golang/go/issues/36641) I believe it will FTBFS on riscv64. Added another commit to avoid this failure.

tianon reviewed (approved):
LGTM

cc @mwhudson :wink:

mwhudson commented:
LGTM, I'll merge this and @lucaskanashiro can then upload? BTW riscv might get cgo pretty soon.

lucaskanashiro commented:
Thanks! I just uploaded it to Groovy. Should I submit a new PR changing the target in `d/changelog` from `UNRELEASED` to `groovy`?

lucaskanashiro commented:
FYI the build failed on riscv64:

https://launchpadlibrarian.net/480602209/buildlog_ubuntu-groovy-riscv64.containerd_1.3.4-0ubuntu1_BUILDING.txt.gz

This is the important part:

```
cd '/<<PKGBUILDDIR>>/.gopath/src/github.com/containerd/containerd' \
	&& make \
		LDFLAGS='' \
		VERSION='1.3.4-0ubuntu1' \
		REVISION='' \
		BUILDTAGS='' \
	&& make man
make[2]: Entering directory '/<<PKGBUILDDIR>>'
+ bin/ctr
-buildmode=pie not supported on linux/riscv64
make[2]: *** [Makefile:187: bin/ctr] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>'
make[1]: *** [debian/rules:26: override_dh_auto_build] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:53: build-arch] Error 2
dpkg-buildpackage: error: debian/rules build-arch subprocess returned exit status 2
```

I am working on it.

tianon commented:
Yeah, a new PR sounds great, thank you!

That build failure is something upstream already has some architecture exclusions in place for, so we might need to either backport a patch, or send a new patch upstream to then backport. 😅

lucaskanashiro commented:
I submitted a patch to fix the build failure on riscv64 and it was already merged:

https://github.com/containerd/containerd/pull/4277

I created an Ubuntu Groovy riscv64 vm locally to test this patch and for some reason the `go env CGO_ENABLED` is not returning `0` as expected:

```
root@ubuntu:~# go env CGO_ENABLED
1
root@ubuntu:~# uname -p
riscv64
root@ubuntu:~# go version
go version go1.14.3 linux/riscv64
```

Due to that I have some test failures. Should I specifically try to check for `GOARCH` and skip tests if it is `riscv64`? Something like this in `debian/rules`:

```
# riscv64 doesn't support cgo
# https://github.com/golang/go/issues/36641
ifeq ($(GOARCH),riscv64)
TAGS += no_btrfs
SKIP += github.com/containerd/containerd/snapshots/btrfs
endif
```
